app/javascript/listings/components/ListingTagsField.jsx

```diff
@@ -0,0 +1,121 @@
+import { useTagsField } from '../../hooks/useTagsField';
+import { TagAutocompleteOption } from '../../article-form/components/TagAutocompleteOption';
+import { TagAutocompleteSelection } from '../../article-form/components/TagAutocompleteSelection';
+import { MultiSelectAutocomplete } from '@crayons';
+
+export const ListingTagsField = ({
+  defaultValue,
+  onInput,
+  categorySlug,
+  name,
+  onFocus,
+}) => {
+  const listingState = useMemo(
+    () => ({
+      additionalTags: {
+        jobs: [
+          'remote',
+          'remoteoptional',
+          'lgbtbenefits',
+          'greencard',
+          'senior',
+          'junior',
+          'intermediate',
+          '401k',
+          'fulltime',
+          'contract',
+          'temp',
+        ],
+        forhire: [
+          'remote',
+          'remoteoptional',
+          'lgbtbenefits',
+          'greencard',
+          'senior',
+          'junior',
+          'intermediate',
+          '401k',
+          'fulltime',
+          'contract',
+          'temp',
+        ],
+        forsale: ['laptop', 'desktopcomputer', 'new', 'used'],
+        events: ['conference', 'meetup'],
+        collabs: ['paid', 'temp'],
+      },
+    }),
+    [],
+  );
+
+  const { defaultSelections, topTags, fetchSuggestions, syncSelections } =
+    useTagsField({ defaultValue, onInput });
+  const [topTagsWithAdditional, setTopTagsWithAdditional] = useState([
+    ...topTags,
+  ]);
+
+  useEffect(() => {
+    if (listingState && listingState.additionalTags) {
+      const newTagsForSelectedCategory =
+        listingState.additionalTags[categorySlug] || [];
+      // Push in this way: { name: 'remote' }
+      const formattedNewTagsForSelectedCategory =
+        newTagsForSelectedCategory.map((name) => ({ name }));
+      setTopTagsWithAdditional([
+        ...topTags,
+        ...formattedNewTagsForSelectedCategory,
+      ]);
+    }
+  }, [topTags, listingState, categorySlug]);
+
+  const fetchSuggestionsWithAdditionalTags = async (searchTerm) => {
+    const suggestionsResult = await fetchSuggestions(searchTerm);
+    const suggestedNames = suggestionsResult.map((t) => t.name);
+
+    // Search in the topTagsWithAdditional array
+    const additionalItems = topTagsWithAdditional.filter((t) =>
+      t.name.startsWith(searchTerm),
+    );
+    // Remove duplicates
+    additionalItems.forEach((t) => {
+      if (!suggestedNames.includes(t.name)) {
+        suggestionsResult.push(t);
+      }
+    });
+    return suggestionsResult;
+  };
+
+  return (
+    <div className="listingform__tagswrapper crayons-field">
+      <MultiSelectAutocomplete
+        defaultValue={defaultSelections}
+        fetchSuggestions={fetchSuggestionsWithAdditionalTags}
+        staticSuggestions={topTagsWithAdditional}
+        staticSuggestionsHeading={
+          <h2 className="crayons-article-form__top-tags-heading">Top tags</h2>
+        }
+        labelText="Tags"
+        showLabel
```

Comment:
(not blocking) - I think this prop defaults to `true` so you could probably omit it here 🙂